frontend/src/components/ChatBox/ChatBox.tsx

```diff
@@ -152,19 +141,13 @@ function ChatBox({
 		// update emails
 		addSentEmails(response.sentEmails);
 
-		if (response.wonLevel && !isLevelComplete()) {
+		if (response.wonLevelMessage) {
 			updateNumCompletedLevels(currentLevel);
-			const successMessage = getSuccessMessage();
-			addChatMessage({
-				type: 'LEVEL_INFO',
-				message: successMessage,
-			});
-			// asynchronously add the message to the chat history
-			void chatService.addInfoMessageToChatHistory(
-				successMessage,
-				'LEVEL_INFO',
-				currentLevel
+			const levelCompleteMessage = makeChatMessageFromDTO(
+				response.wonLevelMessage
```

Comment:
This is maybe something that will emerge after all your changes, but my feeling is that it's fine to return the "wonLevel" flag as your indicator, and the actual "won level" message could arrive as an info message.

Reason being, we already have an optional info message in the response that is added to the chat panel ("transformedMessage" though it's currently a string, not an actual ChatMessage), and so adding another optional field for another info message to the response seems like we are making our API a little too specific. Maybe we want to return a list of (zero or more) info messages to add to the chat?

Likely not something to tackle here, but something to consider as you continue with this work.

Reply:
I like the suggestion. Like, what happens if the message gets transformed, but that message would also win the level? My current solution doesn't support that

Reply:
nevermind, yes it does. But your idea is sleeker

Reply:
that can be a different PR :)

Reply:
https://github.com/orgs/ScottLogic/projects/37/views/2?pane=issue&itemId=57715194 made a ticket for it